src/instance/mod.rs

```diff
@@ -128,3 +166,66 @@ pub trait ToObjectInstance {
 pub trait FromObjectInstance {
     fn from_instance(i: &ObjectInstance) -> Self;
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{parser::Construct, Context, JkInt};
+
+    fn setup() -> Context {
+        let mut ctx = Context::new();
+
+        let inst = Construct::instruction("type Point(x: int, y: int); ")
+            .unwrap()
+            .1;
+        inst.execute(&mut ctx);
+
+        let inst = Construct::instruction("type Vec2(f: Point, s: Point); ")
+            .unwrap()
+            .1;
+        inst.execute(&mut ctx);
+
+        let inst = Construct::instruction("p = Point { x = 1, y = 2}")
+            .unwrap()
+            .1;
+        inst.execute(&mut ctx);
+
+        let inst = Construct::instruction("v = Vec2 { f = p, s = p}")
+            .unwrap()
+            .1;
+        inst.execute(&mut ctx);
+
+        ctx
+    }
+
+    #[test]
+    fn t_one_deep_access() {
+        let mut ctx = setup();
+
+        let inst = Construct::instruction("p").unwrap().1;
+        let p = inst.execute(&mut ctx).unwrap();
+
+        let inst = Construct::instruction("v").unwrap().1;
+        let v = inst.execute(&mut ctx).unwrap();
+        let v_f = v.get_field("f").unwrap();
+        let v_s = v.get_field("s").unwrap();
+
+        assert_eq!(v_f, p);
+        assert_eq!(v_s, p);
+    }
+
+    #[test]
+    fn t_two_deep_access() {
+        let mut ctx = setup();
+
+        let inst = Construct::instruction("v").unwrap().1;
+        let v = inst.execute(&mut ctx).unwrap();
+        let v_f = v.get_field("f").unwrap();
+        let v_s = v.get_field("s").unwrap();
+        let v_f_x = v_f.get_field("x").unwrap();
+        let v_f_y = v_s.get_field("y").unwrap();
+
+        assert_eq!(v_f_x, JkInt::from(1).to_instance());
+        assert_eq!(v_f_y, JkInt::from(2).to_instance());
+    }
+}
```

Comment:
[This](https://doc.rust-lang.org/book/ch11-01-writing-tests.html#using-resultt-e-in-tests) might be an interesting solution to all those `.unwrap()`

Reply:
Oh dang thanks. Yeah I think I'll keep that for another PR though, as the entirety of tests needs to be refactored. I didn't know you could do that